### PR TITLE
docs: resolves markdown error and misspells in "pushing and pulling artifacts" section

### LINKED ISSUE #228: [Documentation Feedback]: wrong sentencing, errors and misspells in docs
### Page URL

https://oras.land/docs/how_to_guides/pushing_and_pulling

### User feedback

Found some misspells, errors and wrong sentencing in the docs, specially the [pushing and pulling](https://oras.land/docs/how_to_guides/pushing_and_pulling) part

anotherissue
I think [OCI Artifacts][artifacts] should be a link, I guess its a markdown error.

myissue

"Pushing single files involves referencing the unique artifact type and at least one file."  maintainers can better rephrase this sentence since they have more knowledge than me, I feel it should be "Pushing artifacts with single file involves referencing the unique artifact type and a file"

anotherbug

"oras should provides a solution" > "oras provides a solution". also the entire para is written in a very confusing way, can be improved

### PR DESCRIPTION
**Pull request by 1Shubham7**
Closes #228

Resolves the 3 errors described in issue #228.

### CODE REVIEW

**Comment by 1Shubham7** [comment]
I made a mistake in the second commit, I misunderstood something, sorry for that, corrected it in the next commit.

**Review by asmitbm — CHANGES_REQUESTED** [review]

File: docs/how_to_guides/pushing_and_pulling.mdx [hunk 1]
```diff
@@ -7,9 +7,9 @@ All users can access container images stored in centralized repositories called
 
 # Pushing
 
-## Pushing artifacts with single files
+## Pushing artifacts with single file
 
-Pushing single files involves referencing the unique artifact type and at least one file.
+Pushing single file artifacts involves referencing the unique artifact type and a file.
```
[the remark this answers is not in the document]
Comment (1Shubham7) [reply docs/how_to_guides/pushing_and_pulling.mdx:12]: but the section is specifically for pushing artifacts with single file, so we would only need one file, not more than one, hence `at least one file` doesn't make sense to me. Please consider it and let me know, final call is obviously yours.

File: docs/how_to_guides/pushing_and_pulling.mdx [hunk 2]
```diff
@@ -72,7 +72,7 @@ In this example, we'll push a collection of files.
 - A single file (`artifact.txt`) that represents overview content that might be displayed as a repository overview
 - A collection of files (`docs/*`) that represents detailed content. When specifying a directory, ORAS will automatically tar the contents.
 
-See [OCI Artifacts][artifacts] for more details.
+See [OCI Artifacts](https://github.com/opencontainers/artifacts/blob/main/artifact-authors.md#defining-a-unique-artifact-type) for more details.
```
Comment (FeynmanZhou) [inline docs/how_to_guides/pushing_and_pulling.mdx:75]:
```suggestion
See [OCI Artifacts](https://github.com/opencontainers/image-spec/blob/main/manifest.md#guidelines-for-artifact-usage) for more details.
```

Comment (FeynmanZhou) [reply docs/how_to_guides/pushing_and_pulling.mdx:75]: https://github.com/opencontainers/artifacts is archived so we need to point it to a new repository.

**Comment by 1Shubham7** [comment]
noted @FeynmanZhou, thanks for helping.

**Comment by 1Shubham7** [comment]
I think it is fine now, changes requested by @asmitbm have been discussed. Please review it @asmitbm .

**Review by FeynmanZhou — DISMISSED** [review]
LGTM

File: docs/how_to_guides/pushing_and_pulling.mdx [hunk 3]
```diff
@@ -140,7 +140,7 @@ oras pull localhost:5000/hello-artifact:v2
 
 ## Using cache when pulling artifacts
 
-In order to save unnecessary network bandwidth and disk I/O oras should provides a solution to pull the artifacts into a local content-address storage (CAS) if the content does not exist, and then copy the artifact to the desired storage.
+In order to save unnecessary network bandwidth and disk I/O, oras provides a solution to pull the artifacts into a local content-address storage (CAS) if the content does not exist, and then copy the artifact to the desired storage.
```
Comment (sajayantony) [inline docs/how_to_guides/pushing_and_pulling.mdx:144]:
```suggestion
Oras pulls the artifact into a content-addressable storage (CAS) cache if the content is not available locally to save bandwidth and disk I/O.
Once the content is availabe in the cache, oras copies the artifact to the desired location. 
```

@Wwwsylvia does the able statement sound accurage. My hope is to split up the long sentence and call out that the cache is a CAS.

Comment (1Shubham7) [reply docs/how_to_guides/pushing_and_pulling.mdx]: thank you @sajayantony for helping

Comment (Wwwsylvia) [reply docs/how_to_guides/pushing_and_pulling.mdx]: LGTM

File: docs/how_to_guides/pushing_and_pulling.mdx [hunk 4]
```diff
@@ -140,8 +140,8 @@ oras pull localhost:5000/hello-artifact:v2
 
 ## Using cache when pulling artifacts
 
-In order to save unnecessary network bandwidth and disk I/O oras should provides a solution to pull the artifacts into a local content-address storage (CAS) if the content does not exist, and then copy the artifact to the desired storage.
-
+Oras pulls the artifact into a content-addressable storage (CAS) cache if the content is not available locally to save bandwidth and disk I/O.
```
Comment (asmitbm) [inline docs/how_to_guides/pushing_and_pulling.mdx:143]:
```suggestion
ORAS pulls the artifact into a content-addressable storage (CAS) cache if the content is not available locally to save bandwidth and disk I/O.
```

Capitalize the word "ORAS" everywhere for consistency.

File: docs/how_to_guides/pushing_and_pulling.mdx [hunk 5]
```diff
@@ -140,8 +140,8 @@ oras pull localhost:5000/hello-artifact:v2
...
+Once the content is availabe in the cache, oras copies the artifact to the desired location.
```
Comment (asmitbm) [inline docs/how_to_guides/pushing_and_pulling.mdx:144]:
```suggestion
Once the content is availabe in the cache, ORAS copies the artifact to the desired location. 
```

**Review by asmitbm — CHANGES_REQUESTED** [review]

**Comment by 1Shubham7** [comment]
added the changes @asmitbm.

**Review by asmitbm — APPROVED** [review]
LGTM🚀

**Review by TerryHowe — APPROVED** [review]
/lgtm

**Pull request merged**